rascal-vscode-extension/src/lsp/JsonOutputChannel.ts

```diff
@@ -0,0 +1,190 @@
+            && "level" in json && typeof json.level === "string"
+            && "message" in json && typeof json.message === "string"
+            && "threadName" in json && typeof json.threadName === "string"
+            && "loggerName" in json && typeof json.loggerName === "string") {
+            return new LogMessage(new Date(json.timestamp), json.level.toUpperCase() as LogLevel, json.message, json.threadName, json.loggerName);
+        }
+
+        return undefined;
+    }
+}
+
+export class JsonParserOutputChannel implements vscode.OutputChannel {
+    readonly name: string;
+
+    private readonly logChannel: vscode.LogOutputChannel;
+
+    constructor(name: string) {
+        this.logChannel = vscode.window.createOutputChannel(name, {log: true});
+        this.name = name;
+    }
+
+    getLogChannel() {
+        return this.logChannel;
+    }
+
+    private printLogOutput(loglevel: LogLevel, message: string) {
+        switch (loglevel) {
+            case LogLevel.fatal: // intentional fall-trough
+            case LogLevel.error: {
+                this.logChannel.error(message);
+                break;
+            }
+            case LogLevel.warn: {
+                this.logChannel.warn(message);
+                break;
+            }
+            case LogLevel.info: {
+                this.logChannel.info(message);
+                break;
+            }
+            case LogLevel.debug: {
+                this.logChannel.debug(message);
+                break;
+            }
+            case LogLevel.trace: {
+                this.logChannel.trace(message);
+                break;
+            }
+            default: {
+                this.logChannel.appendLine(`${loglevel} ${message}`);
+            }
+        }
+    }
+
+    private printPayload(payload: string): void {
+        for (const line of payload.trim().split("\n")) {
+            try {
+                const res = this.printJsonPayLoad(line);
+                if (res === undefined) {
+                    // it's JSON, but we don't understand the shape
+                    this.logChannel.appendLine(line);
+                }
+            } catch (e) {
+                if (e instanceof SyntaxError) {
+                    // this was not JSON at all
+                    this.printNonJsonPayload(line);
+                } else {
+                    this.logChannel.appendLine(`Error while logging ${line}: ${e}`);
+                    throw e;
+                }
+            }
+        }
+    }
+
+    private formatServerTime(date: Date | string): string {
+        if (typeof date === "string") {
+            date = new Date(date);
+        }
+        return date.toISOString();
+    }
+
+    private formatMessage(thread: string, originalTime: Date | string, message: string, loggerName?: string) {
+        const loggerPart = loggerName ? ` ${loggerName}` : "";
+        return `[${thread}] (${this.formatServerTime(originalTime)})${loggerPart} ${message}`;
```

Comment:
it's a bit weird to print both times in a way right? could we for ourselves find a small way to print this? say epoch time?